BuddySave/SharedSaveOrchestrator.cs

```diff
@@ -32,10 +42,21 @@ public void Save(GameSave gameSave)
             _clientNotifier.Notify("There's no lock. Cannot save.");
             return;
         }
-        
-        _clientNotifier.Notify("Uploading game save to cloud...");
-        _cloudManager.UploadSave(gameSave);
-        _cloudManager.DeleteLock(gameSave);
-        _clientNotifier.Notify("Game save lock released.");
+
+        try
+        {
+            _clientNotifier.Notify("Uploading game save to cloud...");
+            _cloudManager.UploadSave(gameSave);
+            _clientNotifier.Notify("Game save uploaded.");
+        }
+        catch (Exception)
+        {
+            _clientNotifier.Notify("Upload failed.");
+        }
+        finally
+        {
+            _cloudManager.DeleteLock(gameSave);
```

Comment:
I wonder if a lock should stay in a case of this failure. Imagine someone is hosting a game, and the save was not shared. Next time someone else wants to host the game will be reverted to the old save.

I think in a case of failure here we need either manual resolution, or some sort of retry option.

Reply:
Good point. I was also thinking about this. 

I am planning to adding lock author and checking it. That could help us solve without manual handling. What do you think?

Reply:
Yeah, I would like a UserName or something in lock.json. Not sure how that helps with handling Save exceptions?

AAhh, as in the same user can remove the lock? That sounds great!